actions/internal/switch-references/switch-references.sh

```diff
@@ -0,0 +1,114 @@
+        echo "$@"
+    fi
+}
+
+show_help() {
+    echo "Usage: $0 [-v|--verbose] <repository_name> <replacement_string> <paths...>"
+    echo ""
+    echo "Replaces all 'uses' references for the specified repository"
+    echo "from @<current_version> to @<replacement_string> into each of the"
+    echo "specified paths."
+    echo ""
+    echo "If a path is a directory, all .yml/.yaml files will be processed"
+    echo "recursively. If a path is a file, only that file will be processed."
+    echo ""
+    echo "Arguments"
+    echo "  -v: verbose"
+    echo ""
+    echo "Examples:"
+    echo "  $0 grafana/plugin-ci-workflows main .github/workflows          # Process directory recursively"
+    echo "  $0 grafana/plugin-ci-workflows v2.0.0 .github/workflows/ci.yml # Process single file"
+    echo "  $0 grafana/plugin-ci-workflows main examples actions/plugins   # Process multiple directories"
+}
+
+
+# Function to process a single file
+process_file() {
+    local file="$1"
+    
+    echov "Processing: $file"
+
+    # Check if file contains the pattern before modifying
+    if grep -q "uses: $REPO_NAME.*@" "$file"; then
+        # Use sed to replace the pattern in-place while preserving comments
+        # Pattern explanation:
+        # - \(uses: $REPO_NAME[^@]*\) - Capture everything up to @
+        # - @[^ ]* - Match @ and everything up to the first space (or end of line)
+        # - \(.*\) - Capture everything after the version (including comments)
+        # Replace with: captured prefix + @REPLACEMENT + captured suffix
+        sed -i "s|\(uses: $REPO_NAME[^@]*\)@[^ ]*\(.*\)|\1@$REPLACEMENT\2|g" "$file"
+        echo "  ✓ Updated $file"
+    else
+        echo "  - No matching pattern found in $file"
+    fi
+}
+
+# Function to process YAML files
+process() {
+    local path="$1"
+
+    # Check if path is a directory (treat as recursive search)
+    if [[ -d "$path" ]]; then
+        echov "Processing directory: $path"
+        
+        # Find all .yml and .yaml files in the directory and subdirectories
+        find "$path" -type f \( -name "*.yml" -o -name "*.yaml" \) -print0 | while IFS= read -r -d '' file; do
+            process_file "$file"
+        done
+    elif [[ -f "$path" ]]; then
+        # Check if it's a YAML file
+        if [[ "$path" == *.yml || "$path" == *.yaml ]]; then
+            process_file "$path"
+        else
+            echov "File $path is not a YAML file, skipping..."
+        fi
+    else
+        echo "Path $path does not exist, skipping..."
+    fi
+}
+
+# Arguments parsing
+VERBOSE=false
+paths=()
+for arg in "$@"; do
+    case $arg in
+        -h|--help) show_help; exit 0 ;;
+        -v|--verbose) VERBOSE=true; shift ;;
+        *)
+            if [ -z "${REPO_NAME:-}" ]; then
+                REPO_NAME="$arg"
+            elif [ -z "${REPLACEMENT:-}" ]; then
+                REPLACEMENT="$arg"
+            else
+                paths+=("$arg")
+            fi
+            shift ;;
```

Comment:
should this shift be inside of the case?

Reply:
Yeah that was a leftover from a previous iteration, I have removed it